routing/cross_mwm_index_graph.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "routing/segment.hpp"
+
+#include <vector>
+
+namespace routing
+{
+/// \brief This is an interface for cross mwm routing.
+// @TODO(bykoianko) This interface will have two implementations.
+// * For orsm cross mwm section.
+// * For A* cross mwm section
+class CrossMwmIndexGraph
+{
+public:
+  /// \returns true if |s| is a transition segment. That means the segment lies on
+  /// mwm border and it's possible to leave the mwm of |s| moving by the segment |s|.
+  virtual bool IsBorder(Segment const & s) const = 0;
```

Comment:
It should be investigated more but as far as I'm aware for the time being in current osrm cross mwm routing:
 * mwm border splits osm road segments in two parts;
 * this two parts are a pair of transition segments. One of them is an exit (from one mwm) segment and another one is an enter (to neighboring mwm) segment.
 * this way for ever mwm there's a set of entering and exiting transition segments. For every transition segment in mwm1 there's (and only one) segment in  mwm2 which sharing the border with mwm1.
 * this scheme now it implemented for osrm cross mwm routing.

Now the method IsBorder is changed:
```
  /// \brief A transition segments is a segment which is located at mwm border and
  /// moving along it the mwm of such segment may be leaved. There are two types of
  /// such segments:
  /// * enter transition segments are segments which are enters to theirs mwm;
  /// * exit transition segments are segments which are exits from theirs mwm;
  /// \returns true if |s| is a exit (|isOutgoing| == true) or enter (|isOutgoing| == false)
  /// transition segment.
  virtual bool IsBorder(Segment const & s, bool isOutgoing) const = 0;
```
> start point of the segment lies on the border, end point doesn't

It's an enter transition point. It's should be kept in cross mwm section. IsBorder(suchSegment, false) returns true.

> start point of the segment doesn't lie on the border, end point does

It's an exit transition point. It's should be kept in cross mwm section. IsBorder(suchSegment, true) returns true.

> both points lie on the border

This case is possible. Probably. But I guess in current implementation of cross mwm routing this case is processed in a special way. To answer I need at first to understand current implementation.

> both points do not lie on the border

There are two possibilities:
* It's not a border segment at all.
* This (osm) segment is crossing the border. If that case it's split in to segments by our generator (as far as I'm aware)